src/threading.c

```diff
@@ -800,28 +800,23 @@ void jl_start_threads(void)
     uv_barrier_init(&thread_init_done, nthreads);
 
     // GC/System threads need to be after the worker threads.
-    int nworker_threads = nthreads - ngcthreads;
+    int nmutator_threads = nthreads - ngcthreads;
 
     for (i = 1; i < nthreads; ++i) {
-        jl_threadarg_t *t = (jl_threadarg_t *)malloc_s(sizeof(jl_threadarg_t)); // ownership will be passed to the thread
-        t->tid = i;
-        t->barrier = &thread_init_done;
-        if (i < nworker_threads) {
+        if (i < nmutator_threads) {
+            jl_threadarg_t *t = (jl_threadarg_t *)malloc_s(sizeof(jl_threadarg_t)); // ownership will be passed to the thread
+            t->tid = i;
+            t->barrier = &thread_init_done;
             uv_thread_create(&uvtid, jl_threadfun, t);
             if (exclusive) {
                 mask[i] = 1;
                 uv_thread_setaffinity(&uvtid, mask, NULL, cpumasksize);
                 mask[i] = 0;
             }
         }
-        else if (i == nthreads - 1 && jl_n_sweepthreads == 1) {
-            uv_thread_create(&uvtid, jl_concurrent_gc_threadfun, t);
-        }
-        else {
-            uv_thread_create(&uvtid, jl_parallel_gc_threadfun, t);
-        }
         uv_thread_detach(&uvtid);
     }
+    jl_spawn_gc_threads();
```

Comment:
I think it's better if this is be called from where `jl_start_threads` is called, to create a clear separation between mutator threads and GC threads.